source/common/common/logger.cc

```diff
@@ -58,14 +58,28 @@ DelegatingLogSinkPtr DelegatingLogSink::init() {
   return delegating_sink;
 }
 
-void Registry::initialize(spdlog::level::level_enum log_level, const std::string& log_format,
-                          Thread::BasicLockable& lock) {
-  // TODO(jmarantz): I think it would be more robust to push a separate lockable
-  // SinkDelegate onto the stack for the lifetime of the lock, so we don't crash
-  // if we try to log anything after the context owning the lock is destroyed.
```

Comment:
The reason I wrote this PR is to try to address this comment about lingering pointers to stale locks. See also #4417 which attempts to work around this in a different way. This may be related to test issues in #4396 though I did not immediately see that connection.